Sources/PackageGraph/Pubgrub/PubgrubDependencyResolver.swift

```diff
@@ -1465,16 +1461,20 @@ private final class ContainerProvider {
     func prefetch(containers identifiers: [PackageReference]) {
         // Process each container.
         for identifier in identifiers {
-            if self.prefetches[identifier] != nil {
-                continue
+            var needsFetching = false
+            self.prefetches.memoize(identifier) {
```

Comment:
How do these DispatchGroups get released (unmemoized?) if they are no longer needed, e.g. if a dependency is removed?  I'm not sure of the cost of a DispatchGroup (I'm guessing it's actually fairly light since it's not associated with any system resources), so it might not be a problem if it stays around for the duration of the workspace even if the dependency is removed.

Reply:
it will be released when `PubgrubDependencyResolver` is released, which seems to be after resolution ends in `Workspace`. speaking of which, the purpose of `activeResolver` in `Workspace` is unclear to me since its reset every time it used?

Reply:
Got it.

Based on Git history it seems to have been added in order to support cancellation, but it wasn't finished:
```
    public func cancelActiveResolverOperation() {
        // FIXME: Need to add cancel support.
    }
```

We will still need that and have a Radar to do it, so I wouldn't remove it at this point.